### PR TITLE
feat: add `airflow.clusterDomain` value

### PR DESCRIPTION
**Pull request by alexkasatikov**
## What issues does your PR fix?

- fixes https://github.com/airflow-helm/charts/issues/476

## What does your PR do?

- Adds a `airflow.clusterDomain` value (default: `cluster.local`) to allow setting the Kubelet `clusterDomain` to non-standard values.

### For all Pull Requests

- [x] Commits are [signed off](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#sign-your-work)
- [x] Commits have [semantic messages](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#semantic-commit-messages)
- [x] Documentation [updated](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#documentation)
- [x] Passes [ct linting](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#linting)

### For releasing ONLY

- [ ] Chart.yaml [version bumped](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#versioning)
- [ ] CHANGELOG.md updated

### CODE REVIEW

File: charts/airflow/values.yaml [hunk 1]
```diff
@@ -232,6 +232,10 @@ airflow:
   ##
   extraVolumes: []
 
+  ## kubernetes cluster domain name configured in the kubelet with clusterDomain option
```
Comment (thesuperzapper) [inline charts/airflow/values.yaml:235]: I think this docstring is more clear if it's worded like:
```
## kubernetes cluster domain name (configured in the kubelet with `--cluster-domain` flag)
```

File: charts/airflow/README.md [hunk 2]
```diff
@@ -1459,6 +1459,7 @@ Parameter | Description | Default
 `airflow.extraContainers` | extra containers for the airflow Pods | `[]`
 `airflow.extraVolumeMounts` | extra VolumeMounts for the airflow Pods | `[]`
 `airflow.extraVolumes` | extra Volumes for the airflow Pods | `[]`
+`airflow.clusterDomain` | kubernetes cluster domain name | `cluster.local`
```
Comment (thesuperzapper) [inline charts/airflow/README.md:1462]: Keep this the same as the docstring:
```
kubernetes cluster domain name (configured in the kubelet with `--cluster-domain` flag)
```

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
@alexkasatikov thanks for the PR, just a small nitpick on the docstring, otherwise looks good!

File: charts/airflow/values.yaml (hunk 1 above)
Comment (alexkasatikov) [reply charts/airflow/values.yaml:235]: Agree but I found that this flag is deprecated and should be moved to configuration file as described here: https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/

Comment (alexkasatikov) [reply charts/airflow/values.yaml]: How about docstring from my second commit?

**Comment by thesuperzapper** [comment]
Bumping for the bot, we still plan to merge this as part of the `8.6.0` release.

**Review by thesuperzapper — APPROVED** [review]
This PR is ready for `8.6.0`, and will be merged as that release is cut.

**Pull request merged**